### DISCUSSION TITLE
Decouple web interface from wardriver logic

### DISCUSSION THREAD

**Discussion by JosephHewitt** [JosephHewitt/wardriver_rev3#119] [Ideas]
Right now the web interface is tightly coupled to the wardriver logic, which is hard to maintain and not at all elegant. I propose the following change and would like feedback regarding this:

- Create a separate folder within this repo for a webapp. Instead of there being any HTML/CSS/JS on the wardriver itself, it will all be in this folder and the wardriver will simply have an API which the webapp can talk to in order to control the wardriver.

This will significantly improve the maintainability of the web interface, allowing for quicker development and therefore likely improved UX/UI.

This webapp still needs to be stored on the wardriver so it can be sent to your web browser whenever you connect. The best way I can currently think of is to distribute the webapp through the OTA update service so it can be an automatic process and so it can also be kept up to date easily.

This method does mean that in order to connect to the wardriver, you will need to either manually put the webapp onto the SD card or connect your wardriver to the internet when you first set it up (the WiFi setup interface will still work using the old method; not the new webapp so this is still possible). This is likely to decrease user satisfaction with the wardriver since it is an extra hurdle to jump over during setup.

Any thoughts regarding this idea? Perhaps alternative ways of implementing it?

**Comment by pejacoby** [comment]
I would say this is the right way to go - decouple the code from the interface. To the degree that it add a set up step or two, it doesn't seem overly onerous, and no more difficult than adding the "Cells" data to the SD card.  With the direct Wigle upload functionality and the OTA update processes in web interface, if it's documented that this is a powerful addition to the rig, I believe most of the user base would work through it.

I'm assuming that In the initial setup process, once the wardriver is on a network it would download the web app folder and give the user a link to the main page?  Subsequent access to the wardriver over http after boot-up would instantiate the interface.

**Reply by JosephHewitt** [reply]
> I'm assuming that In the initial setup process, once the wardriver is on a network it would download the web app folder and give the user a link to the main page? Subsequent access to the wardriver over http after boot-up would instantiate the interface.

Yes, the current first time setup interface would be unchanged which allows people to set up a WiFi network. Once a network is configured, it would download the required files automatically and then load the interface in the browser of the person setting up the device. 

Future usage of the wardriver will not require internet connectivity if the files are already downloaded.
